staging/src/k8s.io/dynamic-resource-allocation/cel/cache_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCacheSemantic(t *testing.T) {
+	// Cache two entries.
+	cache := NewCache(2)
+
+	// Successful compilations get cached.
+	resultTrue := cache.GetOrCompile("true")
+	require.Nil(t, resultTrue.Error)
+	resultTrueAgain := cache.GetOrCompile("true")
+	if resultTrue != resultTrueAgain {
```

Comment:
Interface instances can be compared in Go as if you compared the underlying values or pointers. This can lead to runtime errors when those are not comparable, but that isn't a problem here.

Reply:
Not a wrong answer, but not directly applicable here. The results are structs, not interfaces. Structs that contain only comparable fields are equal if all fields are equal. *That* applies here. Because some of those fields are pointers, we get unique structs for each compilation. Some other fields are interfaces, so my previous answer wasn't completely besides the point. :grin:

Reply:
That was the only option I considered, but it's completely not obvious. Consider adding a comment.

Reply:
[Done](https://github.com/kubernetes/kubernetes/compare/cdb3b2400f371f9b875abfa4f2e9cc7e9b66bd3c..e2edeaafd6dfce1f2ad036562e97c09dc21332bf).

There goes my LGTM label... :sweat_smile: